File: app/src/main/java/com/autopreference/sample/screens/UserPreferenceFragment.java
package com.autopreference.sample.screens;

import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.annotation.Nullable;
import android.support.v4.app.Fragment;
import android.text.Editable;
import android.text.TextWatcher;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.CheckBox;
import android.widget.CompoundButton;
import android.widget.EditText;

import com.autopreference.sample.ApplicationInstance;
import com.autopreference.sample.R;
import com.autopreferences.build.UserPreferences;

public final class UserPreferenceFragment extends Fragment {

	public static final String TAG = "UserPreferenceFragment";
	private final UserPreferences userPreferences = ApplicationInstance.getInstance().getAppPreferences().getUserPreferences();

	@NonNull
	public static UserPreferenceFragment newInstance() {
		Bundle args = new Bundle();
		UserPreferenceFragment fragment = new UserPreferenceFragment();
		fragment.setArguments(args);
		return fragment;
	}

	@Override
	public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
		return inflater.inflate(R.layout.fragment_user_preference, container, false);
	}

	@Override
	public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
		super.onViewCreated(view, savedInstanceState);

		CheckBox userEnabled = view.findViewById(R.id.userEnabled);
		userEnabled.setChecked(userPreferences.isEnabled());
		userEnabled.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
			@Override
			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
				userPreferences.putEnabled(isChecked);
			}
		});

		EditText userName = view.findViewById(R.id.userName);
		userName.setText(userPreferences.getName());
		userName.addTextChangedListener(new TextWatcher() {
			@Override
			public void beforeTextChanged(CharSequence s, int start, int count, int after) {
			}

			@Override
			public void onTextChanged(CharSequence s, int start, int before, int count) {
			}

			@Override
			public void afterTextChanged(Editable s) {
				userPreferences.putName(s.toString());
			}
		});

		EditText userId = view.findViewById(R.id.userId);
		userId.setText(String.valueOf(userPreferences.getId()));
		userId.addTextChangedListener(new TextWatcher() {
			@Override
			public void beforeTextChanged(CharSequence s, int start, int count, int after) {
			}

			@Override
			public void onTextChanged(CharSequence s, int start, int before, int count) {
			}

			@Override
			public void afterTextChanged(Editable s) {
				String toString = s.toString();
				if (toString.isEmpty()) {
					userPreferences.removeId();
				} else {
					userPreferences.putId(Integer.parseInt(toString));
				}
			}
		});
	}
}

File: app/src/main/java/com/autopreference/sample/screens/UserPreferenceFragment.kt
package com.autopreference.sample.screens

import android.os.Bundle
import android.text.Editable
import android.text.TextWatcher
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.CheckBox
import android.widget.EditText
import androidx.fragment.app.Fragment
import com.autopreference.sample.ApplicationInstance
import com.autopreference.sample.R
import ru.starksoft.autopreferences.build.UserPreferences

class UserPreferenceFragment : Fragment() {
    private val userPreferences: UserPreferences =
        ApplicationInstance.instance.appPreferences.getUserPreferences()

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View {
        return inflater.inflate(R.layout.fragment_user_preference, container, false)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        val userEnabled = view.findViewById<CheckBox>(R.id.userEnabled)
        userEnabled.isChecked = userPreferences.isEnabled
        userEnabled.setOnCheckedChangeListener { buttonView, isChecked ->
            userPreferences.putEnabled(
                isChecked
            )
        }

        val userName = view.findViewById<EditText>(R.id.userName)
        userName.setText(userPreferences.name)
        userName.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
            }

            override fun afterTextChanged(s: Editable) {
                userPreferences.putName(s.toString())
            }
        })

        val userId = view.findViewById<EditText>(R.id.userId)
        userId.setText(userPreferences.id.toString())
        userId.addTextChangedListener(object : TextWatcher {
            override fun beforeTextChanged(s: CharSequence, start: Int, count: Int, after: Int) {
            }

            override fun onTextChanged(s: CharSequence, start: Int, before: Int, count: Int) {
            }

            override fun afterTextChanged(s: Editable) {
                val toString = s.toString()
                if (toString.isEmpty()) {
                    userPreferences.removeId()
                } else {
                    userPreferences.putId(toString.toInt())
                }
            }
        })
    }

    companion object {

        const val TAG: String = "UserPreferenceFragment"

        fun newInstance(): UserPreferenceFragment {
            val args = Bundle()
            val fragment = UserPreferenceFragment()
            fragment.arguments = args
            return fragment
        }
    }
}
